File: app/domain/entities/tipoimpuesto.py
from typing import Optional, List
from dataclasses import dataclass, field
from app.domain.entities.enums import (
    TipoAplicacionEnum, BaseTributarioEnum, AmbitoImpuestoEnum, 
    CategoriaImpuestoEnum, TipoUsoImpuestoEnum, MetodoCalculoImpuestoEnum,
    AmbitoUsoImpuestoEnum, CategoriaFiscalImpuestoEnum
)
from app.domain.entities.condiciontributaria import CondicionTributaria
from app.domain.entities.tipo_impuesto_distribucion import TipoImpuestoDistribucion

@dataclass
class TipoImpuesto:
    id: Optional[int] = None
    codigo_afip: str = ""
    nombre: str = ""
    descripcion: str = ""
    tipo_aplicacion: TipoAplicacionEnum = TipoAplicacionEnum.SUMA
    base_calculo: BaseTributarioEnum = BaseTributarioEnum.SUBTOTAL
    ambito: AmbitoImpuestoEnum = AmbitoImpuestoEnum.NACIONAL
    categoria: CategoriaImpuestoEnum = CategoriaImpuestoEnum.IMPUESTO_DIRECTO
    porcentaje: Optional[float] = None
    editable: bool = True
    obligatorio: bool = False
    activo: bool = True
    
    # Odoo fields
    tipo_uso: TipoUsoImpuestoEnum = TipoUsoImpuestoEnum.VENTAS
    metodo_calculo: MetodoCalculoImpuestoEnum = MetodoCalculoImpuestoEnum.PORCENTAJE
    ambito_uso: AmbitoUsoImpuestoEnum = AmbitoUsoImpuestoEnum.AMBOS
    importe: float = 0.0
    etiqueta_factura: Optional[str] = None
    incluido_precio: bool = False
    afecta_base_subsecuente: bool = False
    categoria_fiscal: Optional[CategoriaFiscalImpuestoEnum] = None
    notas_legales: Optional[str] = None
    cuenta_impuesto_vta: Optional[str] = None
    cuenta_impuesto_com: Optional[str] = None
    
    condiciones_asociadas: List[CondicionTributaria] = field(default_factory=list)
    reparticiones: List[TipoImpuestoDistribucion] = field(default_factory=list)

File: frontend/src/domain/entities/TipoImpuesto.ts

import type { TipoAplicacionEnum } from '../enums/TipoAplicacionEnum';
import type { BaseTributarioEnum } from '../enums/BaseTributarioEnum';
import type { AmbitoImpuestoEnum } from '../enums/AmbitoImpuestoEnum';
import type { CategoriaImpuestoEnum } from '../enums/CategoriaImpuestoEnum';
import type { TipoUsoImpuestoEnum } from '../enums/TipoUsoImpuestoEnum';
import type { MetodoCalculoImpuestoEnum } from '../enums/MetodoCalculoImpuestoEnum';
import type { AmbitoUsoImpuestoEnum } from '../enums/AmbitoUsoImpuestoEnum';
import type { CategoriaFiscalImpuestoEnum } from '../enums/CategoriaFiscalImpuestoEnum';
import type { CondicionTributaria } from './CondicionTributaria';
import type { TipoImpuestoDistribucion } from './TipoImpuestoDistribucion';

export interface TipoImpuesto {
    id: number;
    codigo_afip: string;
    nombre: string;
    descripcion: string;
    tipo_aplicacion: TipoAplicacionEnum;
    base_calculo: BaseTributarioEnum;
    ambito: AmbitoImpuestoEnum;
    categoria: CategoriaImpuestoEnum;
    porcentaje?: number;
    editable: boolean;
    obligatorio: boolean;
    activo: boolean;

    // Odoo fields
    tipo_uso: TipoUsoImpuestoEnum;
    metodo_calculo: MetodoCalculoImpuestoEnum;
    ambito_uso: AmbitoUsoImpuestoEnum;
    importe: number;
    etiqueta_factura?: string;
    incluido_precio: boolean;
    afecta_base_subsecuente: boolean;
    categoria_fiscal?: CategoriaFiscalImpuestoEnum;
    notas_legales?: string;
    cuenta_impuesto_vta?: string;
    cuenta_impuesto_com?: string;

    condiciones_asociadas?: CondicionTributaria[];
    reparticiones?: TipoImpuestoDistribucion[];
}

